components/style/values/computed/counters.rs

```diff
@@ -4,11 +4,202 @@
 
 //! Computed values for counter properties
 
+#[cfg(feature = "servo")]
+use computed_values::list_style_type::T as ListStyleType;
+use cssparser::{self, Parser, Token};
+use parser::{Parse, ParserContext};
+use selectors::parser::SelectorParseErrorKind;
+use std::fmt::{self, Write};
+use style_traits::{CssWriter, ParseError, StyleParseErrorKind, ToCss};
+#[cfg(feature = "gecko")]
+use values::generics::CounterStyleOrNone;
 use values::generics::counters::CounterIncrement as GenericCounterIncrement;
 use values::generics::counters::CounterReset as GenericCounterReset;
+#[cfg(feature = "gecko")]
+use values::specified::Attr;
+use values::specified::url::SpecifiedUrl;
+pub use values::specified::{Content, ContentItem};
 
 /// A computed value for the `counter-increment` property.
 pub type CounterIncrement = GenericCounterIncrement<i32>;
 
 /// A computed value for the `counter-increment` property.
 pub type CounterReset = GenericCounterReset<i32>;
+
+impl Content {
+    /// Set `content` property to `normal`.
+    #[inline]
+    pub fn normal() -> Self {
+        Content::Normal
+    }
+
+    #[cfg(feature = "servo")]
+    fn parse_counter_style(
+        input: &mut Parser
+    ) -> ListStyleType {
+        input.try(|input| {
+            input.expect_comma()?;
+            ListStyleType::parse(input)
+        }).unwrap_or(ListStyleType::Decimal)
+    }
+
+    #[cfg(feature = "gecko")]
+    fn parse_counter_style(
+        context: &ParserContext,
+        input: &mut Parser
+    ) -> CounterStyleOrNone {
+        input.try(|input| {
+            input.expect_comma()?;
+            CounterStyleOrNone::parse(context, input)
+        }).unwrap_or(CounterStyleOrNone::decimal())
+    }
+}
+
+impl Parse for Content {
+    // normal | none | [ <string> | <counter> | open-quote | close-quote | no-open-quote |
+    // no-close-quote ]+
+    // TODO: <uri>, attr(<identifier>)
+    fn parse<'i, 't>(
+        context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<Self, ParseError<'i>> {
+        if input.try(|input| input.expect_ident_matching("normal")).is_ok() {
+            return Ok(Content::Normal)
+        }
+        if input.try(|input| input.expect_ident_matching("none")).is_ok() {
+            return Ok(Content::None)
+        }
+        if input.try(|input| input.expect_ident_matching("-moz-alt-content")).is_ok() {
+            #[cfg(feature = "gecko")]
+            return Ok(Content::MozAltContent)
+        }
+
+        let mut content = vec![];
+        loop {
+            match input.try(|i| SpecifiedUrl::parse(context, i)) {
+                #[cfg(feature = "gecko")]
+                Ok(mut url) => {
+                    url.build_image_value();
+                    content.push(ContentItem::Url(url));
+                    continue;
+                }
+                _ => (),
+            }
```

Comment:
It seems this pattern matching is only used for `Ok` and avoiding compiled in servo.
How about wrapping the original `if let` inside `#[cfg(feature = "gecko"]`?

Ex.
```rust
#[cfg(feature = "gecko")] {
  if let Ok(mut url) = input.try(|i| SpecifiedUrl::parse(context, i)) {
    // other code here
  }
}
```

Reply:
Same as above, we can't use use `#[cfg]` attributes on `if` blocks.

Doesn't look very elegant, I agree. Wonder if there's some other better way to do it.

Reply:
Same as before, if you wrap it in a block you can use it.